### ISSUE TITLE
Infinite Loading When Switching Sections

### ISSUE THREAD

**Issue by j0rdyknight** [modrinth/theseus#346]
### Describe the bug

Infinite Loading bug between menus sometimes, after changing instance options, or navigating through menus.

### Steps to reproduce

1. open app
2. jump into instance 
3. options
4. add an icon (this is the one step i use to recreate it, but there are others)
5. click library
6. green bar at the top just loads infinetly and nothing loads.
7. can also click home here, and it wont load eaither 

### Expected behavior

Obviously this is still in Alpha so bugs are expected, i am in no way moaning about this, just wanting to bring it to the attention of the devs, so they can investigate and repair the issue. 

But this infinte loaing shouldnt happnen. Menus should load, and you should be able to navigate and change through the menus with no issue. 

Closing and reopening the program does this, untill it happens agian.

**Comment by maxomatic458** [comment]
im experiencing the same
weirdly that doesnt seem to happen when running theseus in debug mode

thats the error in the devtools console:
```
index-d4c70adb.js:1  RangeError: Maximum call stack size exceeded
    at x.each (index-d4c70adb.js:29:16883)
    at x.truncate (index-d4c70adb.js:29:19102)
    at index-d4c70adb.js:29:19130
    at x.each (index-d4c70adb.js:29:16904)
    at x.truncate (index-d4c70adb.js:29:19102)
    at index-d4c70adb.js:29:19130
    at x.each (index-d4c70adb.js:29:16904)
    at x.truncate (index-d4c70adb.js:29:19102)
    at index-d4c70adb.js:29:19130
    at x.each (index-d4c70adb.js:29:16904)
Z0 @ index-d4c70adb.js:1
```

**Comment by maxomatic458** [comment]
this line here throws the exception:
https://github.com/modrinth/theseus/blob/1f478ec9fc086e6626a9ca9dd7cb290b48000440/theseus_gui/src/pages/Browse.vue#L264

**Comment by CodexAdrian** [comment]
The problem specifically with Theseus hanging on the changing of an icon has been fixed in #298

**Comment by maxomatic458** [comment]
I just cloned the main branch, and i still experience the same thing
here is a clip

https://github.com/modrinth/theseus/assets/104733404/42ec7714-732b-45fa-9636-6205690dba75

looks like the url is only updating on page-reload
